### PR TITLE
generate osgi bundle manifest data for jar #248

### PR DESCRIPTION
**Pull request by mwiede**
changes according to https://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html#_adding_osgi_metadata_to_existing_projects_without_changing_the_packaging_type

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -527,6 +528,20 @@
                 <artifactId>maven-dependency-plugin</artifactId>
                 <version>3.3.0</version>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>5.1.1</version>
```
Comment (norrisjeremy) [inline pom.xml:534]: I think `5.1.8` is the most recent version of this plugin? Any reason to use an older version?

Also, do we need to add an `<extensions>true</extensions>` statement here? I've seen it in other projects, not sure exactly what it does though?

File: pom.xml [hunk 2]
```diff
@@ -527,6 +528,20 @@
...
+                <executions>
+                    <execution>
+                    <id>bundle-manifest</id>
+                    <phase>process-classes</phase>
+                    <goals>
+                        <goal>manifest</goal>
+                    </goals>
+                    </execution>
```
Comment (norrisjeremy) [inline pom.xml:542]: I think we're missing some indentation here between the opening and closing `execution` tags?

File: pom.xml [hunk 3]
```diff
@@ -527,6 +528,20 @@
                 <artifactId>maven-dependency-plugin</artifactId>
                 <version>3.3.0</version>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>5.1.1</version>
+                <executions>
```
Comment (norrisjeremy) [inline pom.xml:535]: Should we add something like this to lock-in that the only public-use package is `com.jcraft.jsch` (similar to what I did in `module-info.java` for JPMS)?
```
                <configuration>
                    <instructions>
                        <Export-Package>com.jcraft.jsch</Export-Package>
                    </instructions>
                </configuration>
```

File: pom.xml (hunk 2 above)
Comment (mwiede) [reply pom.xml:547]: fixed

File: pom.xml (hunk 1 above)
Comment (mwiede) [reply pom.xml]: I have compared the output in the manifest with and without the `extensions` tag and there are no differences.

File: pom.xml [hunk 4]
```diff
@@ -527,6 +528,25 @@
                 <artifactId>maven-dependency-plugin</artifactId>
                 <version>3.3.0</version>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>5.1.8</version>
+                <configuration>
+                    <instructions>
+                        <Export-Package>com.jcraft.jsch.*</Export-Package>
```
Comment (norrisjeremy) [inline pom.xml:537]: I think we just want to export the `com.jcraft.jsch` package?

**Review by norrisjeremy — COMMENTED** [review]
I was keeping the order of the maven plugins approximately in order of the phases in which their specified goals were executed.
Therefore, I would put `maven-bundle-plugin` after `maven-compiler-plugin` and before `scala-maven-plugin` since the `process-classes` phase happens after the `compile` phase but before the `test-compile` phase.

**Review by norrisjeremy — APPROVED** [review]

**Pull request merged**